development/external-plugins/needs-tws/plugin.go

```diff
@@ -0,0 +1,305 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/kyma-project/test-infra/development/prow/externalplugin"
+	"go.uber.org/zap"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/repoowners"
+)
+
+const (
+	DefaultNeedsTwsLabel         = "do-not-merge/missing-docs-review"
+	DefaultTechnicalWritersGroup = "technical-writers"
+)
+
+var markdownRe = regexp.MustCompile("^.*\\.md$")
+
+func HelpProvider(_ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	ph := &pluginhelp.PluginHelp{
+		Description: "needs-tws checks if the Pull Request has modified Markdown files and blocks the merge until it is reviewed and approved by one of the Technical Writers.",
+	}
+	return ph, nil
+}
+
+type githubClient interface {
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+	AddLabel(org, repo string, number int, label string) error
+	RemoveLabel(org, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsCollaborator(org, repo, user string) (bool, error)
+	AssignIssue(org, repo string, number int, logins []string) error
+	BotUserChecker() (func(string) bool, error)
+}
+
+type ownersAliases interface {
+	LoadOwnersAliases(l *zap.SugaredLogger, basedir, filename string) (repoowners.RepoAliases, error)
+}
+
+type AliasesClient struct {
+	ownersAliases
+}
+
+func (o AliasesClient) LoadOwnersAliases(l *zap.SugaredLogger, basedir, filename string) (repoowners.RepoAliases, error) {
+	l.Debug("Load OWNERS_ALIASES")
+	path := filepath.Join(basedir, filename)
+	if _, err := os.Stat(path); err != nil {
+		return nil, err
+	}
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return repoowners.ParseAliasesConfig(b)
+}
+
+type reviewCtx struct {
+	author, issueAuthor, body, htmlURL string
+	number                             int
+	repo                               github.Repo
+	assignees                          []github.User
+	approved                           bool
+}
+
+type PluginBackend struct {
+	ghc githubClient
+	gcf git.ClientFactory
+	oac ownersAliases
+}
+
+func (p *PluginBackend) HandlePR(ep *externalplugin.Plugin, e externalplugin.Event) {
```

Comment:
Does it make sens to have two HandlePR and handlePullRequest methods? Why we need this separation? If this methods are doing something different the names should indicate that. If they do the same logic, why this is not one method.

Reply:
Renamed to `PullRequestHandler`.